### ISSUE TITLE
Style/RedundantBegin does not understand trailing `until`

### ISSUE THREAD

**Issue by nertzy** [rubocop-hq/rubocop#8864]
Style/RedundantBegin does not understand this valid example:

```ruby
begin
  do_first_thing
  some_value = do_second_thing
end until some_value
```

--------

## Expected behavior

Rubocop should not flag this as an error

## Actual behavior

It is flagged by Style/RedundantBegin.

Using auto-correct gives:

```ruby

  do_first_thing
  some_value = do_second_thing
 until some_value

```

```
$ bundle exec rubocop --debug tmp/foo.rb
For /Users/dev/projects/syrup: configuration from /Users/dev/projects/syrup/.rubocop.yml
Inheriting configuration from /Users/dev/projects/groundwork-style/rubocop.yml
Default configuration from /Users/dev/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rubocop-0.93.0/config/default.yml
Inheriting configuration from /Users/dev/projects/groundwork-style/rubocop_performance.yml
configuration from /Users/dev/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rubocop-performance-1.8.1/config/default.yml
configuration from /Users/dev/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rubocop-performance-1.8.1/config/default.yml
Inheriting configuration from /Users/dev/projects/groundwork-style/rubocop_rails.yml
configuration from /Users/dev/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rubocop-rails-2.8.1/config/default.yml
configuration from /Users/dev/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rubocop-rails-2.8.1/config/default.yml
Inheriting configuration from /Users/dev/projects/groundwork-style/rubocop_rake.yml
configuration from /Users/dev/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rubocop-rake-0.5.1/config/default.yml
configuration from /Users/dev/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rubocop-rake-0.5.1/config/default.yml
Inheriting configuration from /Users/dev/projects/groundwork-style/rubocop_rspec.yml
configuration from /Users/dev/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rubocop-rspec-1.43.2/config/default.yml
configuration from /Users/dev/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rubocop-rspec-1.43.2/config/default.yml
Inheriting configuration from /Users/dev/projects/groundwork-style/rubocop_thread_safety.yml
configuration from /Users/dev/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rubocop-thread_safety-0.4.1/config/default.yml
configuration from #{path}
Inheriting configuration from /Users/dev/projects/groundwork-style/rubocop_i18n.yml
.rubocop.yml: RSpec/ExampleLength:Exclude overrides the same parameter in /Users/dev/projects/groundwork-style/rubocop_rails.yml
.rubocop.yml: RSpec/MultipleExpectations:Exclude overrides the same parameter in /Users/dev/projects/groundwork-style/rubocop_rails.yml
.rubocop.yml: AllCops:Include overrides the same parameter in /Users/dev/projects/groundwork-style/rubocop_rspec.yml
.rubocop.yml: Rails/SquishedSQLHeredocs:Enabled overrides the same parameter in /Users/dev/projects/groundwork-style/rubocop_rspec.yml
.rubocop.yml: Style/MethodCalledOnDoEndBlock:Enabled overrides the same parameter in /Users/dev/projects/groundwork-style/rubocop_i18n.yml
Inspecting 1 file
Scanning /Users/dev/projects/syrup/tmp/foo.rb
C

Offenses:

tmp/foo.rb:3:1: C: Style/RedundantBegin: Redundant begin block detected.
begin
^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
Finished in 0.48552500002551824 seconds
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ bundle exec rubocop -V
0.93.0 (using Parser 2.7.1.5, rubocop-ast 0.7.1, running on ruby 2.7.2 x86_64-darwin19)```
```

**Comment by nertzy** [comment]
Thanks! 😄
